### PR TITLE
+ sqlgg 20231201

### PR DESCRIPTION
**Pull request by ygrek**

### CODE REVIEW

File: packages/sqlgg/sqlgg.20231201/opam [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs "@install" "@doc" {with-doc}]
+]
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0"}
+  "menhir"
+  "mybuild"
+  "ppx_deriving"
```
Comment (haochenx) [inline packages/sqlgg/sqlgg.20231201/opam:18]: This should help fixing some of the lower bound failures like [lower-bounds (failed: No ppx driver were found. It seems that ppx_deriving.std is not)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/d280e6363d8e0147c11e8272c24c4a288704c0ea/variant/compilers,4.07,sqlgg.20231201,lower-bounds)

```suggestion
  "ppx_deriving" {>= "4.3"}
```

File: packages/sqlgg/sqlgg.20231201/opam [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+tags: [ "org:ygrek" ]
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs "@install" "@doc" {with-doc}]
+]
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0"}
+  "menhir"
+  "mybuild"
```
Comment (haochenx) [inline packages/sqlgg/sqlgg.20231201/opam:17]: It seems that mybuild does not like the following distros and are failing the CI jobs.
But those are not directly related to this PR and I think we can safely ignore them.

- arm32-ocaml-5.1
- ocaml-5.1-afl
- ppc64-ocaml-5.1
- x86_32-ocaml-5.1

**Review by haochenx — COMMENTED** [review]
can you look into the lower bound check failures like the following?
- [lower-bounds (failed: The function 'Mybuild.Version.save' has type string -> unit)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/d280e6363d8e0147c11e8272c24c4a288704c0ea/variant/compilers,5.1,sqlgg.20231201,lower-bounds)

File: packages/sqlgg/sqlgg.20231201/opam (hunk 2 above)
Comment (mseri) [inline packages/sqlgg/sqlgg.20231201/opam:17]:
```suggestion
  "mybuild" {>= "3"}
```
Fixes
```

#=== ERROR while compiling sqlgg.20231201 =====================================#
# context              2.2.0~alpha4~dev | linux/x86_64 | ocaml-base-compiler.5.0.0 | pinned(https://ygrek.org/p/release/sqlgg/sqlgg-20231201.tar.gz)
# path                 ~/.opam/5.0/.opam-switch/build/sqlgg.20231201
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p sqlgg -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/sqlgg-7-fbec42.env
# output-file          ~/.opam/log/sqlgg-7-fbec42.out
### output ###
# (cd _build/default && /home/opam/.opam/5.0/bin/ocamlc.opt -strict-sequence -strict-formats -short-paths -keep-locs -w +A@8-e-40-41-42-44-45-48-70 -g -bin-annot -I src/.gen_version.eobjs/byte -I /home/opam/.opam/5.0/lib/mybuild -I /home/opam/.opam/5.0/lib/ocaml/unix -I /home/opam/.opam/5.0/lib/ocamlbuild -I /home/opam/.opam/5.0/lib/ppx_deriving/runtime -I /home/opam/.opam/5.0/lib/result -no-alias-deps -o src/.gen_version.eobjs/byte/dune__exe__Gen_version.cmo -c -impl src/gen_version.pp.ml)
# File "src/gen_version.ml", line 3, characters 2-22:
# 3 |   Mybuild.Version.save ~identify:false file
#       ^^^^^^^^^^^^^^^^^^^^
# Error: This function has type string -> unit
#        It is applied to too many arguments; maybe you forgot a `;'.
# (cd _build/default && /home/opam/.opam/5.0/bin/menhir lib/sql_parser.mly --base lib/sql_parser --infer-read-reply lib/sql_parser__mock.mli.inferred)
# Warning: 19 states have shift/reduce conflicts.
# Warning: 3 states have reduce/reduce conflicts.
# Warning: 109 shift/reduce conflicts were arbitrarily resolved.
# Warning: 17 reduce/reduce conflicts were arbitrarily resolved.
```

File: packages/sqlgg/sqlgg.20231201/opam [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+tags: [ "org:ygrek" ]
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs "@install" "@doc" {with-doc}]
+]
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0"}
+  "menhir"
+  "mybuild" {>= "3"}
```
Comment (mseri) [inline packages/sqlgg/sqlgg.20231201/opam:17]:
```suggestion
  "mybuild" {> "3"}
```
Oops

File: packages/sqlgg/sqlgg.20231201/opam [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+license: "GPL-2.0-only"
+tags: [ "org:ygrek" ]
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs "@install" "@doc" {with-doc}]
+]
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0"}
+  "menhir"
```
Comment (mseri) [inline packages/sqlgg/sqlgg.20231201/opam:16]:
```suggestion
  "menhir"  {>= "20180523"}
```
USes `--infer-write-query`

File: packages/sqlgg/sqlgg.20231201/opam [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs "@install" "@doc" {with-doc}]
+]
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0"}
+  "menhir"  {>= "20180523"}
+  "mybuild" {> "3"}
+  "ppx_deriving" {>= "4.3"}
+  ("extlib" | "extlib-compat")
```
Comment (mseri) [inline packages/sqlgg/sqlgg.20231201/opam:19]:
```suggestion
  ("extlib" {>= "1.7.1"} | "extlib-compat" {>= "1.7.1"})
```
This is where `lowercase_ascii` was introduced

**Comment by mseri** [comment]
@ygrek can you make the fixes also upstream?
There is a failure that I still don't understand in https://github.com/ygrek/sqlgg/blob/master/src/gen_caml.ml#L430
```
# File "src/gen_caml.ml", line 438, characters 6-10:
# 438 |     | None -> exec
#             ^^^^
# Error: This pattern should not be a constructor, the expected type is
#        Sqlgg.Sql.param_id
```

**Comment by ygrek** [comment]
the reason is https://github.com/ygrek/ocaml-extlib/commit/4ef8bc2592ea1e3661ea0e7ab57de792998cdfba
ie lower bound for extlib should be 1.7.8
pushed all changes upstream in https://github.com/ygrek/sqlgg/commit/012fb1572b17d9a6864bef8a679a231490d68cdf
thanks!

File: packages/sqlgg/sqlgg.20231201/opam [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0"}
+  "menhir"  {>= "20180523"}
+  "mybuild" {> "3"}
+  "ppx_deriving" {>= "4.3"}
+  ("extlib" {>= "1.7.8"} | "extlib-compat" {>= "1.7.8"})
+  "base-unix"
+  "odoc" {with-doc}
+  "ounit"
```
Comment (shonfeder) [inline packages/sqlgg/sqlgg.20231201/opam:22]: @ygrek Should is this needed outside of tests? Or should it be

```suggestion
  "ounit" {with-test}
```

File: packages/sqlgg/sqlgg.20231201/opam [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0"}
+  "menhir"  {>= "20180523"}
+  "mybuild" {> "3"}
+  "ppx_deriving" {>= "4.3"}
+  ("extlib" {>= "1.7.8"} | "extlib-compat" {>= "1.7.8"})
+  "base-unix"
+  "odoc" {with-doc}
+  "ounit" {with-test}
```
Comment (mseri) [inline packages/sqlgg/sqlgg.20231201/opam:22]:
```suggestion
  "ounit"
```
Needed also for non-test builds, reverting

**Pull request merged**